HW2.5_Best_divider.cpp

```diff
@@ -0,0 +1,30 @@
+#include <cstdint>
+#include <iostream>
+
+int main() {
+  uint32_t input_number{0};
+  std::cerr << "Please, enter your number" << '\n';
+  std::cin >> input_number;
+  int best_divider{0};
+  for (int32_t divider = 1; divider <= input_number; ++divider) {
+    if ((input_number % divider) != 0) {
+      continue;
+    }
+    int div_copy = divider;
+    int best_div_copy = best_divider;
+    int64_t check_number{0};
+    for (; divider > 0; divider /= 10) {
+      check_number += divider % 10;
+      check_number -= best_divider % 10;
+      best_divider /= 10;
+    }
+    divider = div_copy;
+    if (check_number <= 0) {
+      best_divider = best_div_copy;
+    } else {
+      best_divider = div_copy;
+    }
```

Comment:
Этот участок кода можно на порядок проще написать. Я не спорю что он решает задачу, ну поймете что делает этот код когда вернетесь к нему через пол года? или например если кто то другой попытатся его прочитать. Не зная сути задачи я бы так норм поднапрягся.
Вам нужно просто проверить больше 9 делитель, если да, разложить его на цифры получить их сумму и сравнить с максимальным делителем который был
```suggestion
    int64_t check_number{0};
    for (; divider > 0; divider /= 10) {
      check_number += divider % 10;
      check_number -= best_divider % 10;
      best_divider /= 10;
    }
    divider = div_copy;
    if (check_number <= 0) {
      best_divider = best_div_copy;
    } else {
      best_divider = div_copy;
    }
```

Reply:
Мысль не понял. Можете, пожалуйста, привести пример?